### PR TITLE
fix: update reddit_pixel identify call

### PR DESCRIPTION
**Pull request by ItsSudip**
## PR Description

For identify calls we are sending signUp events by default. So we have added support for this event through mapping. Along with that, we have added support for advanced mapping as well.

## Linear task (optional)

Linear task link INT-2063

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
  - Improved user identification for RedditPixel integration with new methods for creating user identifiers and verifying event mappings.
  - Enhanced `loadNativeSdk` function to support conditional initialization based on user identifiers.

- **Bug Fixes**
  - Fixed potential issues with the `identify` method in RedditPixel by adding checks for specific event mappings before executing tracking calls.

- **Tests**
  - Added new tests to validate the functionality of user identifier creation and event mapping verification in the RedditPixel integration.

### CODE REVIEW

**Review by shrouti1507 — COMMENTED** [review]
@ItsSudip can we add some test cases for it.

**Review by yashasvibajpai — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { removeUndefinedAndNullAndEmptyValues } from '../../utils/commonUtils';
+const createUserIdentifier = (traits = {}) => {
+  const userIdentifier = {};
+  const { email, externalId, idfa, aaid } = traits;
+  userIdentifier.email = email;
+  userIdentifier.externalId = externalId;
+  userIdentifier.idfa = idfa;
+  userIdentifier.aaid = aaid;
```
Comment (Gauravudia) [inline packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js:8]: Is it correct to get idfa and aaid from traits? AFAIK we should get this from `context.device `

Comment (shrouti1507) [reply packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js:8]: +1

Comment (ItsSudip) [reply packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js]: I can see we are only using `idfa` from `context.device` [here](https://www.rudderstack.com/docs/sources/event-streams/cloud-apps/appsflyer/#:~:text=idfa-,context.device.advertisingId,-(for%20iOS)). Added that.

**Review by shrouti1507 — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { removeUndefinedAndNullAndEmptyValues } from '../../utils/commonUtils';
+const createUserIdentifier = (traits = {}, context = {}) => {
+  const userIdentifier = {};
+  const { email, externalId, idfa, aaid } = traits;
+  userIdentifier.email = email;
+  userIdentifier.externalId = externalId;
+  userIdentifier.idfa = idfa || context.device?.idfa;
+  userIdentifier.aaid = aaid;
```
Comment (Gauravudia) [inline packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js:8]: I think both should be mapped to `context.device.advertisingId`. Acc. to the doc for reddit pixel
https://business.reddithelp.com/helpcenter/s/article/Reddit-Pixel-Advanced-Matching
```
idfa (Apple’s Identifier for Advertisers) 
aaid (Google Advertising ID)
```

Comment (Gauravudia) [reply packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js:8]: RudderStack support advertisingId in context.device
https://www.rudderstack.com/docs/event-spec/standard-events/common-fields/

**Review by Gauravudia — DISMISSED** [review]

**Pull request merged**